### PR TITLE
ENH Update code to reflect changes in template layer

### PR DESCRIPTION
**Pull request by GuySartorelli**
Relies and reflects on changes in https://github.com/silverstripe/silverstripe-framework/pull/11405

## Issue
- https://github.com/silverstripe/silverstripe-framework/issues/11322

### CODE REVIEW

File: src/Service/MethodRegistry.php [hunk 1]
```diff
@@ -70,7 +70,7 @@ public function getMethods(): array
             if (!$method instanceof MethodInterface) {
                 throw new UnexpectedValueException(sprintf(
                     'Given method "%s" does not implement %s',
-                    $method,
+                    get_class($method),
```
Comment (ChloeSartorelli) [inline src/Service/MethodRegistry.php:73]: This isn't _necessary_ anymore, since I had to revert the changes to what `ModelData::__toString()` returns - but it was still a really bad idea to assume that will return the class name. I still think _this_ change is objectively good and worthwhile.

**Review by emteknetnz — APPROVED** [review]
